rundeckapp/src/test/groovy/org/rundeck/app/providers/GormUserDataProviderSpec.groovy

```diff
@@ -37,6 +37,37 @@ class GormUserDataProviderSpec extends RundeckHibernateSpec implements DataTest
         "login" | false
     }
 
+    @Unroll
+    def "Find or create User case insensitive "() {
+        given:
+        provider.configurationService = Mock(ConfigurationService) {
+           1 * getBoolean(provider.NAME_CASE_SENSITIVE_ENABLED,false) >> false
+        }
+        def loginName = "loginName1"
+        User savedUser = new User(login: loginName)
+        savedUser.save()
+        when:
+        provider.findOrCreateUser(loginName.toUpperCase())
+        then:
+        User.findAll().size() == 1
+    }
+
+    @Unroll
+    def "Find or create User case sensitive  "() {
```

Comment:
there is no need to have 2 separated tests for this, you should use "where" block to provide the necessary parameters and test both cases in a single test block